### DISCUSSION TITLE
Is it intentional that "role: system" part of API request is not at position 0 of the json object?

### DISCUSSION THREAD

**Discussion by Originalimoc** [danny-avila/LibreChat#5139] [Q&A]
### What happened?

Some API provider will just add the system prompt as a second one instead of replacing the default one defined in Jinja2.

### Steps to Reproduce

Seems long prompt will put the system prompt at second last instead of first? Not 100% reproducible.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
Yes it’s intentional. Since a lot of custom endpoints don’t seem to support this, I’ll change this or at least make it an option to place at index 0 instead of length - 2.

**Comment by Originalimoc** [comment]
What's the benefit?

**Comment by Originalimoc** [comment]
I now ran into it models that just ignore everything before system prompt at (length - 2), where is the code that do this/how can I change it? On latest pre-built docker image.

**Reply by danny-avila** [reply]
I'll add an option within the week. thanks!

**Reply by Lavanille777** [reply]
> I'll add an option within the week. thanks!我将在一周内添加一个选项。谢谢！

Has this option been added?
